aten/src/ATen/native/mkldnn/xpu/detail/QConv.cpp

```diff
@@ -0,0 +1,220 @@
+#include <ATen/native/mkldnn/xpu/detail/oneDNNContext.h>
+
+#include <oneapi/dnnl/dnnl.hpp>
+
+namespace at::native::onednn {
+
+static std::tuple<dnnl::memory::desc, dnnl::memory::desc, dnnl::memory::desc>
+qconv_get_md(
+    const at::Tensor& src,
+    const at::Tensor& wgh,
+    const at::Tensor& dst,
+    int64_t groups,
+    bool is_channels_last_suggested) {
+  // create dnnl::memory desc from the src/wgh/dst tensors
+  dnnl::memory::desc src_usr_md, wgh_usr_md, dst_usr_md;
+  auto ndim = src.ndimension();
+  auto fmt_src = conv_src_fmt(ndim, is_channels_last_suggested);
+
+  auto src_tz = src.sizes().vec();
+  auto src_data_t = get_onednn_dtype(src);
+  src_usr_md = dnnl::memory::desc(src_tz, src_data_t, fmt_src);
+
+  auto dst_tz = dst.sizes().vec();
+  auto dst_data_t = get_onednn_dtype(dst);
+  dst_usr_md = dnnl::memory::desc(dst_tz, dst_data_t, fmt_src);
+
+  auto ic = src.size(1);
+  auto oc = dst.size(1);
+  auto wei_data_t = dnnl::memory::data_type::s8;
+  dnnl::memory::dims wgh_tz =
+      compatible_weight_dims(ndim, groups, oc, ic, wgh.sizes());
+  auto fmt_wgh = conv_weight_fmt(ndim, groups != 1, is_channels_last_suggested);
+  wgh_usr_md = dnnl::memory::desc(wgh_tz, wei_data_t, fmt_wgh);
+
+  return {src_usr_md, wgh_usr_md, dst_usr_md};
+}
+
+at::Tensor quantized_convolution(
+    at::Tensor act,
+    double act_scale,
+    int64_t act_zero_point,
+    at::Tensor weight,
+    at::Tensor weight_scales,
+    at::Tensor weight_zero_points,
+    c10::optional<at::Tensor> bias,
+    torch::List<int64_t> stride,
+    torch::List<int64_t> padding,
+    torch::List<int64_t> dilation,
+    bool transposed,
+    int64_t groups,
+    at::Tensor output,
+    double inv_output_scale,
+    int64_t output_zero_point,
+    c10::optional<at::Tensor> accum,
+    double accum_scale,
+    int64_t accum_zero_point,
+    c10::optional<c10::ScalarType> output_dtype,
+    c10::optional<c10::string_view> binary_attr,
+    c10::optional<at::Scalar> binary_alpha,
+    c10::optional<c10::string_view> unary_attr,
+    torch::List<c10::optional<at::Scalar>> unary_scalars,
+    c10::optional<c10::string_view> unary_algorithm) {
+  // TODO: use arg to create proper attr
+  Attr attr =
+      Attr(/*q_scale=*/1.0 / inv_output_scale, /*zp=*/output_zero_point);
+
+  construct_attr_by_post_op(
+      binary_attr.has_value() ? binary_attr.value() : "none",
+      binary_alpha.has_value() ? binary_alpha.value().to<double>() : 1.0,
+      accum_scale,
+      accum_zero_point,
+      unary_attr.has_value() ? unary_attr.value() : "none",
+      unary_scalars,
+      unary_algorithm.has_value() ? unary_algorithm.value() : "",
+      attr);
+
+  auto ndim = act.ndimension();
+  if (bias.has_value()) {
+    attr.append_bias(bias.value(), ndim - 2);
+  }
+  TORCH_CHECK(
+      3 == ndim || 4 == ndim || 5 == ndim,
+      "convolution only supports 3D, 4D, 5D tensor");
+  TORCH_CHECK(
+      output.defined(), "Quantized convlution should always define output");
```

Comment:
```suggestion
      output.defined(), "A valid output is required for quantized convolution.");
```

Reply:
The code has been modified, thanks for suggestions